packages/special-pages/pages/duckplayer/app/providers/OrientationProvider.jsx

```diff
@@ -1,36 +1,55 @@
-import { h } from "preact";
-import { useContext, useEffect, useState } from "preact/hooks";
-import { createContext } from "preact";
-
-const OrientationContext = createContext(/** @type {"landscape" | "portrait"} */("portrait"))
+import { useEffect } from "preact/hooks";
 
 /**
  * Device orientation
- *
- * @param {Object} props - The props for the settings provider.
- * @param {import("preact").ComponentChild} props.children - The children components to be wrapped by the settings provider.
+ * @param {object} props
+ * @param {(orientation: 'portrait' | 'landscape') => void} props.onChange
  */
-export function OrientationProvider ({ children }) {
-    const [orientation, setTheme] = useState(() => {
-        const initial = window.innerWidth > window.innerHeight ? 'landscape' : 'portrait'
-        return /** @type {"landscape"|"portrait"} */(initial)
-    })
+export function OrientationProvider ({ onChange }) {
+    useEffect(() => {
+        if (!screen.orientation?.type) return;
+        onChange(getOrientationFromScreen())
+        const handleOrientationChange = () => {
+            onChange(getOrientationFromScreen());
+        };
+        screen.orientation.addEventListener('change', handleOrientationChange);
+        return () => screen.orientation.removeEventListener('change', handleOrientationChange);
+    }, []);
 
     useEffect(() => {
-        const listener = (e) => setTheme(window.innerWidth > window.innerHeight ? 'landscape' : 'portrait')
+        let timer;
+        const listener = () => {
+            clearTimeout(timer);
+            timer = setTimeout(() => onChange(getOrientationFromWidth()), 300)
```

Comment:
Why the 300s timeout on screen resize? Is it to allow for the orientation change to conclude?

Reply:
It's just a debounce because 'resize' events are noisy